app/Http/Controllers/Api/Review/ReviewPhotoController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Hedonist\Http\Controllers\Api\Review;
+
+use Illuminate\Support\Facades\Storage;
+use Hedonist\Actions\Review\DeleteReviewPhotoAction;
+use Hedonist\Actions\Review\DeleteReviewPhotoRequest;
+use Hedonist\Actions\Review\SaveReviewPhotoRequest;
+use Hedonist\Actions\Review\SaveReviewPhotoAction;
+use Hedonist\Actions\Review\GetReviewPhotoAction;
+use Hedonist\Actions\Review\GetReviewPhotoRequest;
+use Hedonist\Http\Controllers\Api\ApiController;
+use Hedonist\Http\Requests\Review\ReviewPhotoRequest;
+
+class ReviewPhotoController extends ApiController
+{
+    public $saveReviewPhotoAction;
+    public $getReviewPhotoAction;
+    public $deleteReviewPhotoAction;
+
+    public function __construct(
+        SaveReviewPhotoAction $saveReviewPhotoAction,
+        GetReviewPhotoAction $getReviewPhotoAction,
+        DeleteReviewPhotoAction $deleteReviewPhotoAction
+    ) {
+        $this->saveReviewPhotoAction = $saveReviewPhotoAction;
+        $this->getReviewPhotoAction = $getReviewPhotoAction;
+        $this->deleteReviewPhotoAction = $deleteReviewPhotoAction;
+    }
+
+    public function upload(ReviewPhotoRequest $request)
+    {
+        try {
+            $file = $request->file('img_url');
```

Comment:
$file = $request->file('img_url');
$newFileName = time() . '.' . $file->extension();
$file->storeAs('public/upload/review', $newFileName);
All logic should be inside action. You can pass `$file` from httprequest to SaveReviewPhotoRequest